From 55e0334cf5a43948b65788980cd8ab729427230e Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Thu, 30 May 2019 12:25:29 +0100
Subject: [PATCH 518/520] overlays: Fix mmc-related overlays after refactor

The addition of the mmcnr node to the base dtbs caused some overlays to
not work as they should. Patch up pi3-disable-wifi, balena-fin and
sdhost.

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 arch/arm/boot/dts/overlays/balena-fin-overlay.dts       | 7 ++++---
 arch/arm/boot/dts/overlays/pi3-disable-wifi-overlay.dts | 7 +++++++
 arch/arm/boot/dts/overlays/sdhost-overlay.dts           | 7 +++++++
 3 files changed, 18 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/overlays/balena-fin-overlay.dts b/arch/arm/boot/dts/overlays/balena-fin-overlay.dts
index b10675c02757..8c1cba99fc8f 100644
--- a/arch/arm/boot/dts/overlays/balena-fin-overlay.dts
+++ b/arch/arm/boot/dts/overlays/balena-fin-overlay.dts
@@ -5,13 +5,12 @@
 	compatible = "brcm,bcm2835";
 
 	fragment@0 {
-		target = <&mmc>;
-		sdio_wifi: __overlay__ {
+		target = <&mmcnr>;
+		__overlay__ {
 			pinctrl-names = "default";
 			pinctrl-0 = <&sdio_pins>;
 			bus-width = <4>;
 			brcm,overclock-50 = <35>;
-			non-removable;
 			status = "okay";
 		};
 	};
@@ -43,6 +42,8 @@
 				compatible = "gpio-poweroff";
 				gpios = <&gpio 40 1>;
 				force;
+				pinctrl-names = "default";
+				pinctrl-0 = <&power_ctrl_pins>;
 			};
 
 			i2c_soft: i2c@0 {
diff --git a/arch/arm/boot/dts/overlays/pi3-disable-wifi-overlay.dts b/arch/arm/boot/dts/overlays/pi3-disable-wifi-overlay.dts
index 1d061bdd40f3..75e046463900 100644
--- a/arch/arm/boot/dts/overlays/pi3-disable-wifi-overlay.dts
+++ b/arch/arm/boot/dts/overlays/pi3-disable-wifi-overlay.dts
@@ -10,4 +10,11 @@
 			status = "disabled";
 		};
 	};
+
+	fragment@1 {
+		target = <&mmcnr>;
+		__overlay__ {
+			status = "disabled";
+		};
+	};
 };
diff --git a/arch/arm/boot/dts/overlays/sdhost-overlay.dts b/arch/arm/boot/dts/overlays/sdhost-overlay.dts
index ae6c6ad83076..0b72b4eeac88 100644
--- a/arch/arm/boot/dts/overlays/sdhost-overlay.dts
+++ b/arch/arm/boot/dts/overlays/sdhost-overlay.dts
@@ -22,6 +22,13 @@
 		};
 	};
 
+	fragment@2 {
+		target = <&mmcnr>;
+		__overlay__ {
+			status = "disabled";
+		};
+	};
+
 	__overrides__ {
 		overclock_50     = <&frag0>,"brcm,overclock-50:0";
 		force_pio        = <&frag0>,"brcm,force-pio?";
-- 
2.19.1

